File: OCP/ Kata_ Shops Types/SolucionJava/Shop/Shop/AbstractShop.java
package SolucionJava.Shop.Shop;

public abstract class AbstractShop {
    protected String name;
    protected Integer phone;

    public AbstractShop(String name, Integer phone){
        this.name = name;
        this.phone = phone;
    }

    public String getName(){
        return this.name;
    }

    public Integer getPhone(){
        return this.phone;
    }
}

File: OCP/ Kata_ Shops Types/SolucionPhp/Shop/Shop/AbstractShop.php
<?php

declare(strict_types=1);

namespace SolucionPhp\Shop\Shop;

abstract class AbstractShop
{
    private string $name;
    private int $phone;

    public function __construct(string $name, int $phone)
    {
        $this->name = $name;
        $this->phone = $phone;
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function getPhone(): int
    {
        return $this->phone;
    }
}
